### PR TITLE
Announcements section

### PR DESCRIPTION
**Pull request by Dexmaster**
screenshot-jakarta 20scn com-2020 09 08-18_38_36

P.S. The idea is simple each new announcement can have folder with _index.md or for simple announcements just .md file. Where name represents url. 
(e.g. `/content/announcements/jakarta-ee-9-milestone-released/_index.md` can be transformed to `content/announcements/jakarta-ee-9-milestone-released.md` if we don't need folder, and it will represent same url `/announcements/jakarta-ee-9-milestone-released/`)
file structure

### CODE REVIEW

**Comment by chrisguindon** [comment]
I am converting this as a draft since I don't believe it's ready for review!

Please change the state when you think it's time for us to review the code!

**Comment by Dexmaster** [comment]
I've added .adoc as a part of example (because adoc style is easier to write for some developers than md, and it is supported by hugo), code it was ready for review a week ago...

**Comment by chrisguindon** [comment]
> @chrisguindon I've added .adoc as a part of example (because adoc style is easier to write for some developers than md, and it is supported by hugo), code was ready for review a week ago...

@Dexmaster most of the content is lorem ipisum. We would need to change that before we would want to push this live.

In regards to .adoc, I think this is a brilliant idea. May I recommend that we create a separate issue for this? I think we can add .adoc support independently to the work. We shouldn't need to finalize the work here to add .adoc support to jakarta.ee!

**Comment by aeiras** [comment]
What is the status of this feature? 
Can anyone help with an overview on what can be improved as we move closer to the Jakarta EE 9 final release and will need a formalized process that allows for:

- Marketing Committee (review of content)

- Steering committed (signed off of that content)

- Signature - the Jakarta EE Working Group

- Hosted content on the Announcement page, dropping "crossposting links." 

- Making this page fully policed by the Marketing Committee, so not Member is highlighted or members are dropped from the formal communications.

**Comment by dblevins** [comment]
In issue #880 @chrisguindon points out the [Jakarta.ee Website Ownership](https://docs.google.com/spreadsheets/d/1efy2BjsB4zfxNGIA7ESXL_bgXZKfpfHJlRMt19zzuro/edit#gid=98299565) spreadsheet.  The proposal for the announcements section is it would represent the voice of the Working Group itself.  That'd effectively add a new row to the spreadsheet with an `A` on the Steering Committee and an `R` on the Marketing Committee.  For us to move forward we'd need to bring this before the Steering Committee, show the idea and get agreement to accept that accountability.

**Comment by aeiras** [comment]
Excellent!

NOW,  who is owning that conversation? the Steering committee calls are still private in nature; dedicated members can only add agenda items? :)  

we need this completed before Jakarta EE 9 release arrives. :) 

Today, I sent the Jakarta EE WG forum the PR's announcement, its goal, and what we hope to improve. https://www.eclipse.org/lists/jakarta.ee-wg/msg00480.html  [THREAD]

**Comment by dblevins** [comment]
Presentation given to the Steering Committee today:

 - https://docs.google.com/presentation/d/1gFrPUkq1YVpmVKT7S2VGDRasvSbj_RMgEpqCrQP_cLI/edit?usp=sharing
 - https://deploy-preview-879--jakartaee.netlify.app/announcements/

**Comment by trulymobilenwp** [comment]
A few points related to this request.

1. As discussed during our marketing committee call this aligns with our Q2 plans around reworking key pages on Jakarta.ee and should be addressed as that work is done so the overall look and feel comes out as intended by our overall update work.
2. I recommend that any updates to the announcements page should be done in the existing https://jakarta.ee/news/announcements/ page and/or options as currently available at https://jakarta.ee/news/
3. I recommend we determine the best process to ensure the announcements are agreed to collectively by the committee
4. As far as having announcements actually hosted as part of the Jakarta.ee domain, we should address this as part of our rework of the website but we need to ensure we will have a consistent volume of announcements to justify the work and a reasonable way to implement the support of this work.

**Comment by karenmcnaughton** [comment]
Hi all--we will review this on the committee call. We need to consider how to consolidate this with the newsroom, news, blog etc. And will need content here for it to go live. Also, what are the publishing guidelines for this--needs to be defined & documented...who publishes/how/when/type of content etc.

**Comment by dblevins** [comment]
I've created a PR that deletes the old boilerplate content and adds new content.

 - https://github.com/jakartaee/jakarta.ee/pull/1067
 - https://deploy-preview-1067--jakartaee.netlify.app/announcements/
 
Some of this is copy/paste from our actual announcements and therefore ready to go.  Some of these are posts about recent events we could write and publish when they are ready.

The idea is when we go live there'll be content there so it doesn't look empty.

**Comment by dblevins** [comment]
@Dexmaster we met today to discuss this and it looks like everyone wants to this forward, so we just need a few tweaks before we can merge.  The consensus is that "announcements" will be rebranded as "news", so we should move the section from `jakarta.ee/announcements/` to `jakarta.ee/news/`.  There is a page at `/news/` already, we can delete that and replace that section.  I also notice there's a relative URL bug in the 'See all announcements' link at the bottom of each post, so we should fix that while we're at it.

@chrisguindon can you also take a quick look to see if there are any technical tweaks you'd want?

For all: what we decided today was that we'll want to use the same tech to make two other sections `jakarta.ee/blogs/` and `jakarta.ee/events/`.  We should not do that in this PR, however.  We should get this one merged and open separate PRs when the Marketing Committee gives the green light (has time to pre-fill those sections with some posts).

Thanks again @karenmcnaughton @trulymobilenwp @TanjaObradovic @edbratt and @dominislawa for the collaboration today!

**Comment by chrisguindon** [comment]
> @chrisguindon can you also take a quick look to see if there are any technical tweaks you'd want?

My biggest concern is with the CSS. We are planning to start a website re-design project shortly. 

Can we create a separate less file to encompass all the styles that are needed for this new section? 

I think that will help us with the redesign project.

**Comment by Dexmaster** [comment]
> Can we create a separate less file to encompass all the styles that are needed for this new section?

It adds only 100-150 lines of CSS, which have unique classes on the top level (cannot intersect or influence other pages), I will extract those lines into a new file tomorrow.

Will also change names/path at the same time.

**Comment by Dexmaster** [comment]
Done renaming /announcements/ to /news/ and extracted news.less, also done small refactoring for future.

**Comment by dblevins** [comment]
@chrisguindon looks like the CSS has been addressed.  Can you take another look and give a thumbs up/down from a technical perspective?

@trulymobilenwp Looks like this is good to go for a final signoff from the Marketing Committee.  Once merged, we can create separate PRs for the requested "events" and "blogs" sections.

**Comment by trulymobilenwp** [comment]
@dblevins @Dexmaster From our last Marketing Meeting, this is good to go from a Marketing perspective.

File: less/news.less
Comment (chrisguindon) [inline less/news.less]: I don't believe this line is needed.

File: layouts/partials/news/main_prefix.html
Comment (chrisguindon) [inline layouts/partials/news/main_prefix.html:1]: I do have some concerns with us overriding this file.  This new section will miss out on future updates from the base theme.

The default shortcode allows authors to disable the title, can we leverage that?
https://github.com/EclipseFdn/hugo-solstice-theme/blob/master/layouts/partials/main_prefix.html

We can define Archetypes https://gohugo.io/content-management/archetypes/ for the default values we expect authors to use.

If not, we can merge this as-is but my preference would be to use our default file to make sure that we can pull-in future enhancement from our base theme.

File: Gemfile
Comment (chrisguindon) [inline Gemfile:1]: Where are we using asciidoctor?

File: content/news/2021-new-marketing-members/_index.md
Comment (chrisguindon) [inline content/news/2021-new-marketing-members/_index.md]: We currently do not allow html in markdown files:
https://github.com/EclipseFdn/hugo-solstice-theme#known-issues

However, I am starting to think that this could make things easier for contributors. 

Can someone create a separate issue and ask that we enable this on jakarta.ee if folks are interested in this functionnality?

File: content/news/_index.md
Comment (chrisguindon) [inline content/news/_index.md:3]: I think we want to keep seo_title and description based on the SEO report that was done recently. 

If the content is no longer relevant, I would recommend that we update it instead of removing it.

File: content/news/announcements/2020-01-28-cn4j-day/index.md
Comment (chrisguindon) [inline content/news/announcements/2020-01-28-cn4j-day/index.md:1]: I am concerned that we are deleting content from the site. 

This will break existing content such as https://newsroom.eclipse.org/news/announcements/eclipse-foundation-announces-cn4j-day-2020-event-building-enterprise-java

If we want to remove these pages on the site, the team will need to work with our Marketing/Newsroom team in order to migrate the full content to the newsroom website and remove the redirect to jakarta.ee

Comment (chrisguindon) [reply content/news/announcements/2020-01-28-cn4j-day/index.md:1]: This comment applies to all deleted news items.

File: layouts/partials/news/list/summary.html
Comment (chrisguindon) [inline layouts/partials/news/list/summary.html]: Jakarta.ee is currently available in English and Chinese language.

Because of that requirement, I would like to see the implementation of i18n functions in this update: https://gohugo.io/functions/i18n/

**Review by chrisguindon — CHANGES_REQUESTED** [review]
First of all, I love seeing our community working together to improve our website and I really like the way the page looks!

I have some more feedback for the team, however, I don't think I am raising any controversial issues.

I think we will be ready from a technical point of view once my feedback is done! 

Please let me know if you have any questions!

File: Gemfile
Comment (Dexmaster) [reply Gemfile:1]: We can have articles in .md or .adoc :) Same question was answered in September in this PR :)

File: content/news/_index.md
Comment (Dexmaster) [reply content/news/_index.md:3]: will add seo title 👍

File: layouts/partials/news/list/summary.html
Comment (Dexmaster) [reply layouts/partials/news/list/summary.html]: sure will do now

File: Gemfile
Comment (chrisguindon) [reply Gemfile:1]: Yes, I remember briefly but I was under the impression that we were going to remove it.

Can you please convert 1 page of content as .asciidoc to make sure that this works as expected? I also think we should add a pointers to asciidoc documentation in the README.md if the WG is going to support this on the site now.

FWIW - My preference would be to create a separate issue/pr for this since this can change how people contribute content to the whole site, not just news/announcement. 

However, if we can make sure that this works as expected and we have pointers to the doc. I am fine with including this here.

File: layouts/partials/news/main_prefix.html
Comment (Dexmaster) [reply layouts/partials/news/main_prefix.html:1]: The issue with main_prefix is I need hide_page_title for taxonomy and list pages, for which I need to override current functionality.

As for archetypes will create default `archetypes/news.md` have not found any currently used but I'm glad to be first one.

File: Gemfile
Comment (chrisguindon) [reply Gemfile:1]: To clarify any confusion, I am 100% on board with adding .adoc support!

File: layouts/partials/news/main_prefix.html
Comment (chrisguindon) [reply layouts/partials/news/main_prefix.html:1]: I would expect the site to use our defaults if we don't have one in the Jakarta.ee repository:
https://github.com/EclipseFdn/hugo-solstice-theme/tree/master/archetypes

File: less/news.less
Comment (Dexmaster) [reply less/news.less]: thank you yep, it's not urgent here, just needed colour variables, but because it's not main less file it works :)

**Comment by chrisguindon** [comment]
Are we planning to include the Publishing Guidelines in our README.md with this PR?

I think it's going to be important that we have clear guidelines on 1) HOW to submit content 2) WHO can submit content, 3) What  type of content can be submitted and, 4) who is responsible for approving/reviewing the content before it's merged by a website committer.

File: content/news/2021-new-marketing-members/_index.md
Comment (Dexmaster) [reply content/news/2021-new-marketing-members/_index.md]: will replace for now, with md and limit size on css side

File: layouts/partials/news/main_prefix.html
Comment (Dexmaster) [reply layouts/partials/news/main_prefix.html:1]: it cannot we need categories/tags, authors and other fields which it doesn't have :)

but I will use it as example.

Comment (Dexmaster) [reply layouts/partials/news/main_prefix.html:1]: Provided simple archetype with main fields expected from an article, as to not confuse authors.

We can provide more fields or that link in README...

File: Gemfile
Comment (Dexmaster) [reply Gemfile:1]: I originally had few examples in .adoc, but it was lorem ipsum :) will look into conversion do you want it in this PR?

Comment (Dexmaster) [reply Gemfile:1]: I don't believe it's urgent to show an example, this PR is in the pipes from September, I would prefer doing additional stuff in separate PR.

File: content/news/2021-new-marketing-members/_index.md
Comment (Dexmaster) [reply content/news/2021-new-marketing-members/_index.md]: replacement done

File: content/news/_index.md
Comment (Dexmaster) [reply content/news/_index.md:3]: done

File: layouts/partials/news/list/summary.html
Comment (Dexmaster) [reply layouts/partials/news/list/summary.html]: done

File: less/news.less
Comment (Dexmaster) [reply less/news.less]: done

File: content/news/announcements/2020-01-28-cn4j-day/index.md
Comment (Dexmaster) [reply content/news/announcements/2020-01-28-cn4j-day/index.md:1]: Done old news migration with link aliases

**Comment by Dexmaster** [comment]
@chrisguindon most of the changes you've asked for are done, please check. Thank you!

**Comment by dblevins** [comment]
> Are we planning to include the Publishing Guidelines in our README.md with this PR?
> 
> I think it's going to be important that we have clear guidelines on 1) HOW to submit content 2) WHO can submit content, 3) What type of content can be submitted and, 4) who is responsible for approving/reviewing the content before it's merged by a website committer.

@chrisguindon It'll be the marketing committee drafting those guidelines.  We've discussed it in pretty good detail and have slides that cover some of it, but I'll work with them to draft something in README format.  That would be a separate PR.

**Comment by dblevins** [comment]
@chrisguindon can you do a final review of this and let us know if there are any outstanding changes you want?

**Review by chrisguindon — COMMENTED** [review]
My last nitpick is with the images. They are quite big and we should optimize them for web: https://github.com/jakartaee/jakarta.ee/blob/d694408be7b2c813f556e84538fb4f058b3a535a/content/news/openliberty-jakartaee9-webprofile-compatible/banner.png

That banner is 1.04 MB.

**Comment by dblevins** [comment]
> My last nitpick is with the images. They are quite big and we should optimize them for web: https://github.com/jakartaee/jakarta.ee/blob/d694408be7b2c813f556e84538fb4f058b3a535a/content/news/openliberty-jakartaee9-webprofile-compatible/banner.png
> 
> That banner is 1.04 MB.

Let us know what size you want as the max, or some kind of guideline, and we can update all the images.

**Comment by Dexmaster** [comment]
> My last nitpick is with the images. They are quite big and we should optimize them for web: https://github.com/jakartaee/jakarta.ee/blob/d694408be7b2c813f556e84538fb4f058b3a535a/content/news/openliberty-jakartaee9-webprofile-compatible/banner.png
> 
> That banner is 1.04 MB.

Optimized news images. (not much optimization with width of 1350px, from 10 to 144 kb, but still optimized)

Also added small optimize_images.sh bash script, which is run in `content/news` by default. Careful, it also strips alpha levels which is ok for news but might not be good for all images in an `images` folder, so it's folder specific. It also downsizes all unoptimized images that are larger in width than 1350 px.

P.S. if you prefer some other optimization tools tell me and I can amend/revert my last commit.

**Comment by Dexmaster** [comment]
Rebased to current head.

@chrisguindon as to reiterate my previous comment. All new images optimized in the last commit, also bash script for content/news images optimization included.

**Comment by chrisguindon** [comment]
@Dexmaster Thanks so much! We don't have any specific requirements other than doing the best that we can to optimize our images so that our users can load them faster.

The challenge here is striking a balance between good quality in your image and the file size.

THanks for making those changes!

**Comment by chrisguindon** [comment]
What is the next step for this project? Is there any more approval that needs to get done or is this basically ready to be committed?

**Comment by jgallimore** [comment]
@TanjaObradovic are there any further issues with this, or can this now be merged in?

**Comment by Dexmaster** [comment]
Rebased

**Comment by jgallimore** [comment]
All of the feedback has been incorporated, and, unless I'm mistaken, everyone has agreed that this can be merged, and go live.

Please can we move forward and get this merged?

**Comment by chrisguindon** [comment]
@TanjaObradovic @karenmcnaughton are we ready to push this on prod? I can merge the change if there are no objections!

**Comment by TanjaObradovic** [comment]
Looking at https://deploy-preview-1067--jakartaee.netlify.app/announcements/ it seems to be great! no concerns.

**Comment by chrisguindon** [comment]
Thank all involved! The changes have been merged into the src branch!

**Comment by aeiras** [comment]
Chris and Infra Team, 

With your help, merging this PR is possible. Thank you! 
Merging it is the BEST way to end April as Jakarta EE 9.1 is happening the middle of Mat.  
 Thank you for helping us improve effectively how the Working Group communicates to the world. 

You and team rocks!

**Comment by aeiras** [comment]
my public thank you to you Chris and the team! https://twitter.com/ameliaeiras/status/1388205673020674050?s=20

Hugs

**Pull request merged**
